packages/liveblocks-core/src/comments/comment-body.ts

```diff
@@ -149,6 +148,10 @@ export type StringifyCommentBodyOptions<
   ) => PromiseOrNot<(TUserMeta["info"] | undefined)[] | undefined>;
 };
 
+function isSomething<T>(input: null | undefined | T): input is T {
+  return input !== null && input !== undefined;
+}
```

Comment:
I assume this helper exists purely to be able to filter null elements from an array, and retain that knowledge statically in the types?

Sometimes I use a `.flatMap()` for use cases like this:

```ts
// ❌ Instead of…
someList
  .map(x => condition ? x : null)
  .filter(isSomething);

// ✅ Do a flat map instead
someList
  .flatMap(x => condition ? [x] : []);
```

This effectively does the same thing, and "maps and filters" the elements in a single pass, by either returning a 1-element-list, or an empty list. It is still 100% type safe, and avoids the need for helpers like this.

Reply:
Yep exactly, it's purely here because of TS's `filter` limitations.

Ooh interesting, I'll switch to that!